rosetta/retriever/validator.go

```diff
@@ -15,12 +15,13 @@
 package retriever
 
 import (
+	"github.com/onflow/flow-go/model/flow"
 	"github.com/optakt/flow-dps/rosetta/identifier"
 )
 
 type Validator interface {
-	Account(rosAccountID identifier.Account) error
-	Block(rosBlockID identifier.Block) (identifier.Block, error)
-	Transaction(rosTxID identifier.Transaction) error
-	Currency(rosCurrency identifier.Currency) (identifier.Currency, error)
+	Account(rosAccountID identifier.Account) (address flow.Address, err error)
```

Comment:
Just thinking, it might make sense for the validators to return the same type as the type they are validating..

E.g. if the validator is validating `identifier.Block`, it returns `height` and `flow.Identifier`. It might read nicer if it returned the complete `identifier.Block` instead of its components.

Reply:
That's exactly what I changed, because it made no sense the other way. After validating the Rosetta identifiers, we use native Flow types for the remaining function calls, and we also often replace fields on the Rosetta identifiers before returning them.

It is a lot cleaner to receive the native Flow types from the validator, use those for our function calls, and then build the returned Rosetta identifiers again from the authoritative values in Flow.